Alr.Platforms.Windows: use Alire.config in msys2 install

Fabien-Chouteau opened the pull request.

Fabien-Chouteau committed:
Alr.Platforms.Windows: use Alire.config in msys2 install

Fabien-Chouteau committed:
Update msys2 install script URL

Now that the master version is updated with install dir feature.

Fabien-Chouteau committed:
Alire.Config.Edit: create directory for the config file in case it doesn't exists

mosteo commented on src/alr/os_windows/alr-platforms-windows.adb:
```diff
@@ -149,29 +149,54 @@ package body Alr.Platforms.Windows is
+         --  Save msys2 install dir in the global config
+         Cfg.Edit.Set (Path  => Cfg.Filepath (Cfg.Global),
+                       Key   => "msys2.install_dir",
+                       Value => Alire.Utils.YAML.YAML_Stringify (Install_Dir));
```
I think that the `TOML` library escapes strings as needed. Just curious, has this any advantage?

Fabien-Chouteau replied:
I actually did this because of an error I got. I will check what the TOML lib can do.

Fabien-Chouteau replied:
Should be fixed. Thanks for pointing it out, I was not happy about this YAML call...

Fabien-Chouteau committed:
Alire.Config.Edit: improve TOML value evaluation

If first the value cannot be parsed, we directly create a TOML_String value instead of trying to parse it again between quotes.

Fabien-Chouteau committed:
Alr.Platforms.Windows: remove unecessary call to YAML_Stringify

JeremyGrosser commented:
Oops, wrong source url.
